### ISSUE TITLE
Rustc shows unnecessary big error when glob reexport failed

### ISSUE THREAD

**Issue by WaffleLapkin** [rust-lang/rust#81413]
I tried this code:

```rust
pub const ITEM: Item = Item;

pub struct Item;

pub fn item() {}

// Glob (*) is required to cause the error
pub use doesnt_exist::*;

mod a {
    use crate::{item, Item, ITEM};
}

mod b {
    use crate::item;
    use crate::Item;
    use crate::ITEM;
}

mod c {
    use crate::item;
}
```
([play](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=d7d38220140a6793756d30e20fdbdc22))

I expected to see an error that only says that "`doesnt_exist` is unresolved".

But the error says that all uses of items from the module with glob import (`crate` in this case) are unresolved too:

```text
error[E0432]: unresolved imports `doesnt_exist`, `crate::item`, `crate::Item`, `crate::ITEM`, `crate::item`, `crate::Item`, `crate::ITEM`, `crate::item`
  --> src/lib.rs:41:9
   |
41 | pub use doesnt_exist::*;
   |         ^^^^^^^^^^^^ use of undeclared crate or module `doesnt_exist`
...
44 |     use crate::{item, Item, ITEM};
   |                 ^^^^  ^^^^  ^^^^
...
48 |     use crate::item;
   |         ^^^^^^^^^^^
49 |     use crate::Item;
   |         ^^^^^^^^^^^
50 |     use crate::ITEM;
   |         ^^^^^^^^^^^
...
54 |     use crate::item;
   |         ^^^^^^^^^^^
```

This may lead to extremely long and unreadable errors. The issue was originally found while developing `teloxide` where it caused [1500+ lines error](https://gist.github.com/WaffleLapkin/0283b070e38fed0affdf5b787b0010d3).

### Meta

`rustc --version --verbose`:
```
rustc 1.51.0-nightly (c97f11af7 2021-01-10)
binary: rustc
commit-hash: c97f11af7bc4a6d3578f6a953be04ab2449a5728
commit-date: 2021-01-10
host: x86_64-unknown-linux-gnu
release: 1.51.0-nightly
```

**Comment by SNCPlay42** [comment]
<details>
<summary>Output on 1.38:</summary>

```
error[E0432]: unresolved import `doesnt_exist`
 --> <source>:8:9
  |
8 | pub use doesnt_exist::*;
  |         ^^^^^^^^^^^^ maybe a missing crate `doesnt_exist`?

warning: unused imports: `ITEM`, `Item`, `item`
  --> <source>:11:17
   |
11 |     use crate::{item, Item, ITEM};
   |                 ^^^^  ^^^^  ^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: unused import: `crate::item`
  --> <source>:15:9
   |
15 |     use crate::item;
   |         ^^^^^^^^^^^

warning: unused import: `crate::Item`
  --> <source>:16:9
   |
16 |     use crate::Item;
   |         ^^^^^^^^^^^

warning: unused import: `crate::ITEM`
  --> <source>:17:9
   |
17 |     use crate::ITEM;
   |         ^^^^^^^^^^^

warning: unused import: `crate::item`
  --> <source>:21:9
   |
21 |     use crate::item;
   |         ^^^^^^^^^^^

error: aborting due to previous error
```
</details>

@rustbot label A-diagnostics A-resolve regression-from-stable-to-stable

**Comment by WaffleLapkin** [comment]
@SNCPlay42 why do you think it's a "regression-from-stable-to-stable"? IMO the output from `1.38` is even worse

**Comment by SNCPlay42** [comment]
The *error* is much shorter, and only points at the actual problem. The unused import warnings, of course, wouldn't be there if those imports were actually used.

**Comment by SNCPlay42** [comment]
```
********************************************************************************
Regression in nightly-2019-09-11
********************************************************************************
...
found 6 bors merge commits in the specified range
  commit[0] 2019-09-09UTC: Auto merge of #64313 - Centril:rollup-7w8b67g, r=Centril
  commit[1] 2019-09-10UTC: Auto merge of #64321 - Centril:rollup-jsj5tpl, r=Centril
  commit[2] 2019-09-10UTC: Auto merge of #64333 - Centril:rollup-llhhr82, r=Centril
  commit[3] 2019-09-10UTC: Auto merge of #64329 - Mark-Simulacrum:rustdoc-log, r=GuillaumeGomez
  commit[4] 2019-09-10UTC: Auto merge of #60387 - Goirad:test-expansion, r=ollie27
  commit[5] 2019-09-10UTC: Auto merge of #64354 - Centril:rollup-oaq0xoi, r=Centril
```

From those rollups, #64054 looks relevant.

**Comment by WaffleLapkin** [comment]
@SNCPlay42 ah, you are right, I've misread the output thinking that the warnings were errors too.

**Comment by osa1** [comment]
I spent some time trying to debug this last week and I'm fairly sure #64054 is the PR that introduced this. I wasn't aware of that PR at the time but I spent a lot of time trying to understand what is "determinate" vs. "indeterminate" (which is unfortunately not documented) and I remember finding out that with glob import the import `crate::item` is becoming "indeterminate" while it's normally (without the glob import) it is "determinate", which is one of the things we check elsewhere when generating these warnings.

**Comment by Stupremee** [comment]
Assigning `P-medium` as [discussed as part of the Prioritization Working Group procedure](https://rust-lang.zulipchat.com/#narrow/stream/245100-t-compiler.2Fwg-prioritization.2Falerts/topic/.2381413.20Rustc.20shows.20unnecessary.20big.20error.20when.20glob.20reexport.E2.80.A6/near/224230451) and removing `I-prioritize`.
